great_expectations/dataset/dataset.py

```diff
@@ -107,6 +119,13 @@ def inner_wrapper(
                 column = kwargs.get("column")
 
             if column is not None:
+                if (
```

Comment:
I think we need a comment here about what's going on...this is a roundabout way of testing whether we're one of our known subclass implementations.

Reply:
Totally - I did it this way to avoid circular dependencies, but I'll add some explanation.